Sources/ComposableArchitecture/SwiftUI/NavigationStackStore.swift

```diff
@@ -217,6 +215,16 @@ extension NavigationLink where Destination == Never {
   }
 }
 
+private struct NavigationDestinationView<State, Destination: View>: View {
+  let component: Component<State>
+  let destination: (Component<State>) -> Destination
+  var body: some View {
+    self.destination(component)
+      .environment(\.navigationDestinationType, State.self)
+      .id(component.id)
```

Comment:
Looking at this again, I'm not sure if this `id` modifier should go here or inside the `destination` block of `navigationDestination(...)`? What is this `id` used for @mbrandonw ?